tests/ui/to_string_in_display.stderr

```diff
@@ -0,0 +1,27 @@
+error: Using to_string in fmt::Display implementation might lead to infinite recursion
+  --> $DIR/to_string_in_display.rs:24:25
+   |
+LL |         write!(f, "{}", self.to_string())
+   |                         ^^^^^^^^^^^^^^^^
+   |
+   = note: `-D clippy::to-string-in-display` implied by `-D warnings`
+
+error: type `C` implements inherent method `to_string(&self) -> String` which shadows the implementation of `Display`
+  --> $DIR/to_string_in_display.rs:35:5
+   |
+LL | /     pub fn to_string(&self) -> String {
+LL | |         "I am C".to_string()
+LL | |     }
+   | |_____^
+   |
+   = note: `#[deny(clippy::inherent_to_string_shadow_display)]` on by default
+   = help: remove the inherent method from type `C`
+
+error: Using to_string in fmt::Display implementation might lead to infinite recursion
+  --> $DIR/to_string_in_display.rs:42:25
+   |
+LL |         write!(f, "{}", self.to_string())
```

Comment:
This should not trigger the lint, iff the `C::to_string`, instead of the `<C as ToString>::to_string` method is used. See my comment in: https://github.com/rust-lang/rust-clippy/pull/5574#issuecomment-625333045

Reply:
@flip1995 I tried several utils but stuck to think of the solution to this case... Is there any way to get closer to a solution to this problem? Thanks.

Reply:
You have to check if the method is from the `ToString` trait. You can use code similar to
https://github.com/rust-lang/rust-clippy/blob/799eef691969c2f6ecf33359f0b1e1691a060648/clippy_lints/src/loops.rs#L545-L546

for this. 

- `next`  -> `to_string`
- `match_expr` -> `expr`
- `ITERATOR` -> `TO_STRING`

Reply:
Thank you for your suggestion. I will try this weekend.